code/modules/antagonists/hivemind/hivemind.dm

```diff
@@ -0,0 +1,194 @@
+	if(!silent && owner.current)
+		to_chat(owner.current,"<span class='userdanger'> Your psionic powers fade, you are no longer the hivemind's host! </span>")
+	owner.special_role = null
+	..()
+
+/datum/antagonist/hivemind/proc/add_objective(var/datum/objective/O)
+	objectives += O
+
+/datum/antagonist/hivemind/proc/remove_objective(var/datum/objective/O)
+	objectives -= O
+
+/datum/antagonist/hivemind/proc/forge_objectives()
+	if(prob(65))
+		var/datum/objective/hivemind/hivesize/size_objective = new
+		size_objective.owner = owner
+		add_objective(size_objective)
+	else
+		var/datum/objective/hivemind/hiveescape/hive_escape_objective = new
+		hive_escape_objective.owner = owner
+		add_objective(hive_escape_objective)
+	if(prob(50))
+		var/datum/objective/hivemind/assimilate/assim_objective = new
+		assim_objective.owner = owner
+		if(prob(25)) //Decently high chance to have to assimilate an implanted crew member
+			assim_objective.find_target_by_role(pick("Captain","Head of Security","Security Officer","Detective","Warden"))
+		if(!assim_objective.target) //If the prob doesn't happen or there are no implanted crew, find any target
+			assim_objective.find_target()
+		assim_objective.update_explanation_text()
+		add_objective(assim_objective)
+	else if(prob(70))
+		var/datum/objective/assassinate/kill_objective = new
+		kill_objective.owner = owner
+		kill_objective.find_target()
+		add_objective(kill_objective)
+	else
+		var/datum/objective/maroon/maroon_objective = new
+		maroon_objective.owner = owner
+		maroon_objective.find_target()
+		add_objective(maroon_objective)
+
+	var/datum/objective/escape/escape_objective = new
+	escape_objective.owner = owner
+	add_objective(escape_objective)
+
+	return
+
+/datum/antagonist/hivemind/greet()
+	to_chat(owner.current, "<B><font size=3 color=red>You are the host of a powerful Hivemind.</font></B>")
+	to_chat(owner.current, "<b>Your psionic powers will grow by assimilating the crew into your hive. Use the Assimilate Vessel spell on a stationary \
+		target, and after ten seconds he will be one of the hive. This is completely silent and safe to use, and failing will reset the cooldown. As \
+		you assimilate the crew, you will gain more powers to use. Most are silent and won't help you in a fight, but grant you great power over your \
+		vessels. There are other hiveminds onboard the station, collaboration is possible, but a strong enough hivemind can reap many rewards from a \
+		well planned betrayal.</b>")
+	owner.current.playsound_local(get_turf(owner.current), 'sound/ambience/antag/tatoralert.ogg', 100, FALSE, pressure_affected = FALSE)
+
+	owner.announce_objectives()
+
+/datum/antagonist/hivemind/roundend_report()
+	var/list/result = list()
+
+	var/greentext = TRUE
+
+	result += printplayer(owner)
+	result += "<b>Hive Size:</b> [hive_size]"
+
+	var/objectives_text = ""
+	if(objectives.len)
+		var/count = 1
+		for(var/datum/objective/objective in objectives)
+			if(objective.check_completion())
```

Comment:
Use printobjectives instead of writing the whole thing by hand.

Reply:
Done